packages/client/src/workflow-client.ts

```diff
@@ -0,0 +1,580 @@
+import os from 'os';
+import { temporal } from '@temporalio/proto';
+import { DataConverter, defaultDataConverter, Workflow } from '@temporalio/workflow';
+import { WorkflowSignalType, WorkflowQueryType } from '@temporalio/workflow/lib/interfaces';
+import { WorkflowClientInterceptors } from './interceptors';
+import { v4 as uuid4 } from 'uuid';
+import { composeInterceptors } from '@temporalio/workflow';
+import { nullToUndefined } from '@temporalio/workflow/lib/time';
+import { arrayFromPayloads, mapToPayloads } from '@temporalio/workflow/lib/converter/data-converter';
+import * as errors from '@temporalio/workflow/lib/errors';
+import { WorkflowOptions, CompiledWorkflowOptions, addDefaults, compileWorkflowOptions } from './workflow-options';
+import {
+  WorkflowClientCallsInterceptor,
+  WorkflowQueryInput,
+  WorkflowSignalInput,
+  WorkflowTerminateInput,
+} from './interceptors';
+import {
+  StartWorkflowExecutionRequest,
+  GetWorkflowExecutionHistoryRequest,
+  DescribeWorkflowExecutionResponse,
+  TerminateWorkflowExecutionResponse,
+  RequestCancelWorkflowExecutionResponse,
+} from './types';
+import { Connection, WorkflowService } from './connection';
+
+type EnsurePromise<T> = T extends Promise<any> ? T : Promise<T>;
+
+/// Takes a function type F and converts it to an async version if it isn't one already
+type AsyncOnly<F extends (...args: any[]) => any> = (...args: Parameters<F>) => EnsurePromise<ReturnType<F>>;
+
+export interface WorkflowClientOptions {
+  /**
+   * {@link DataConverter} to use for serializing and deserializing payloads
+   */
+  dataConverter?: DataConverter;
+
+  /**
+   * Used to override and extend default Connection functionality
+   *
+   * Useful for injecting auth headers and tracing Workflow executions
+   */
+  interceptors?: WorkflowClientInterceptors;
+
+  /**
+   * Identity to report to the server
+   *
+   * @default `${process.pid}@${os.hostname()}`
+   */
+  identity?: string;
+
+  /**
+   * Identity to report to the server
+   *
+   * @default `${process.pid}@${os.hostname()}`
+   */
+  namespace?: string;
+
+  /**
+   * Should a query be rejected by closed and failed workflows
+   *
+   * @default QUERY_REJECT_CONDITION_UNSPECIFIED which means that closed and failed workflows are still queryable
+   */
+  queryRejectCondition?: temporal.api.enums.v1.QueryRejectCondition;
+}
+
+export type WorkflowClientOptionsWithDefaults = Required<WorkflowClientOptions>;
+
+export function defaultWorkflowClientOptions(): WorkflowClientOptionsWithDefaults {
+  return {
+    dataConverter: defaultDataConverter,
+    // ManagementFactory.getRuntimeMXBean().getName()
```

Comment:
Da heck

Reply:
The equivalent in java (where I copied it from), I'll just clarify